flowershop/models/stock_warehouse.py

```diff
@@ -0,0 +1,105 @@
+from odoo import fields,models
+from odoo.exceptions import UserError,ValidationError
+import requests
+from datetime import datetime,time
+
+
+
+class Stockwarehouse(models.Model):
+    
+    _inherit = 'stock.warehouse'
+ 
+
+    weather_ids = fields.One2many('stock.warehouse.weather',
+    'weather_rec',
+        string='weather')
+
+    def _get_api_key_and_location(self, show_error):
+        api_key = self.env["ir.config_parameter"].sudo().get_param("flower_shop.weather_api_key")
+
+        if api_key == "unset" or not api_key:
+
+            raise ValidationError("API KEY DOES NOT EXIST")
+
+        if not self.partner_id or not self.partner_id.partner_latitude or not self.partner_id.partner_longitude:
+
+            raise ValidationError(f"FAILED TO GET LAT AND LONG FOR WAREHOUSE :{self.name}" )
+
+        return api_key, self.partner_id.partner_latitude, self.partner_id.partner_longitude
```

Comment:
Why do you have show_error in the function parameters if you are not using  in the function. check the solution to see a better approach. 

What you have done here is not wrong, just a better non blocking approach can be done